plugin/src/main/kotlin/com/varabyte/kobweb/intellij/startup/KobwebPostStartupProjectActivity.kt

```diff
@@ -0,0 +1,25 @@
+package com.varabyte.kobweb.intellij.startup
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.externalSystem.service.project.manage.ProjectDataImportListener
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+import com.varabyte.kobweb.intellij.services.project.KobwebProjectCacheService
+
+/**
+ * Actions to perform after the project has been loaded.
+ */
+class KobwebPostStartupProjectActivity : ProjectActivity {
+    private class ImportListener(private val project: Project) : ProjectDataImportListener {
+        override fun onImportFinished(projectPath: String?) {
+            // After an import / gradle sync, let's just clear the cache, which should get automatically rebuilt
+            // as users interact with their code.
+            project.service<KobwebProjectCacheService>().clear()
+        }
+    }
+
+    override suspend fun execute(project: Project) {
+        val messageBusConnection = project.messageBus.connect()
+        messageBusConnection.subscribe(ProjectDataImportListener.TOPIC, ImportListener(project))
+    }
+}
```

Comment:
Do we really need this? I don't see why we would clear the cache when it:  
A: Is already cleared, because we just spun up the IDE instance
B: We import the project but the cache is still valid.

The only reason I see for clearing the cache is when we change something with gradle, which should then be put in the KobwebProjectResolver. I think there are also some methods one can override that get run when the project is finished/starting syncing

Reply:
I believe this is a fairly standard way to register a listener that gets triggered on project syncing. If you find another way, please explicitly recommend what you found so we don't have to back and forth extra rounds.

I skimmed the AbstractProjectResolverExtension APIs and it doesn't look like there's any obvious method to override. It's possible there's one to override that's triggered as a side effect of a sync, but clarity is important here too.

The code listener registered above is a clear expression of my intention. I'm happy to change it if you find another one that doesn't require as many lines of code.

Some people leave their IntelliJ window open for days and days, during which they may sync upstream code multiple times. We really want to clear our cache out when we can. A stale cache is a debugging nightmare.

Reply:
If you think thats for the best then sure.
Here i found a method that gets run before syncing:   void preImportCheck();

Reply:
Yeah, I saw that method. If they had called it `preImport` I might have used it. I didn't like that it was called `preImportCheck` though. That implied it's meant more for verifying, not for side effects.

I think there's also a bit of value in decoupling the cache logic from something that is Gradle specific.

As a side note, when I review people's code that doesn't sit right with me, I ask myself the following:

* if we figure out a better way later, will it be really hard to refactor? (I will spend way more time resisting a public API going into a library for example than a private method going into an application module)
* is their solution really some one-off weird approach that's hard to understand? Or is there precedent for that approach elsewhere in the codebase?
* do I know of anything better?

I'm definitely not sure my way is the best way. However, I do think it is relatively isolated, so if we find a better way later, it should be trivial to migrate.

And when I search online how to add listeners to IntelliJ, a lot of old docs recommend using IntelliJ components, which have since been deprecated and replaced with Project Activity (https://plugins.jetbrains.com/docs/intellij/plugin-components.html#project-open). So it's not like I'm using some bizarre mad scientist approach here as far as I can tell!

Reply:
I guess you are right about that. Its really not a lot of code